### PR TITLE
Add FAB base class and set import_name explicitly.

### PR DESCRIPTION
**Pull request by jhtimmins**
This change was previously made and then reverted bc it caused static assets to load incorrectly.. The bug has been fixed by explicitly passing the name "flask_appbuilder.base" into the blueprint.

This is part of the ongoing epic to refactor Airflow's auth architecture. Currently, the auth code includes methods that are no longer needed given how Airflow uses auth, but they're still required by FABs base class. By copying the base class directly into Airflow, with attribution to Daniel Gaspar, FAB's author, we can modify the base class as necessary to simplify Airflow's auth code.

In addition to the blueprint name, the only change is customizing the behavior of `. _check_and_init()`.

Tl;Dr -- Copies FAB's AppBuilder class directly into Airflow.

### CODE REVIEW

**Review by bbovenzi — APPROVED** [review]
Tested locally. UI is now working as expected.

File: airflow/www/extensions/init_appbuilder.py [hunk 1]
```diff
@@ -15,11 +15,627 @@
...
+        update_perms=conf.getboolean('webserver', 'UPDATE_FAB_PERMS'),
+    ):
+        """
+        AppBuilder constructor
+        :param app:
+            The flask app object
+        :param session:
+            The SQLAlchemy session object
+        :param menu:
+            optional, a previous contructed menu
```
Comment (uranusjr) [inline airflow/www/extensions/init_appbuilder.py]:
```suggestion
            optional, a previous constructed menu
```

File: airflow/www/extensions/init_appbuilder.py [hunk 2]
```diff
@@ -15,11 +15,627 @@
...
+            You can use an url string or an endpoint name
+            if non provided default_view from view will be set as href.
+        :param icon:
+            Font-Awesome icon name, optional.
+        :param label:
+            The label that will be displayed on the menu,
+            if absent param name will be used
+        :param category:
+            The menu category where the menu will be included,
+            if non provided the view will be acessible as a top menu.
```
Comment (uranusjr) [inline airflow/www/extensions/init_appbuilder.py]:
```suggestion
            if non provided the view will be accessible as a top menu.
```

File: airflow/www/extensions/init_appbuilder.py [hunk 3]
```diff
@@ -15,11 +15,627 @@
...
+        menu=None,
+        indexview=None,
+        base_template='airflow/main.html',
+        static_folder="static/appbuilder",
+        static_url_path="/appbuilder",
+        security_manager_class=None,
+        update_perms=conf.getboolean('webserver', 'UPDATE_FAB_PERMS'),
+    ):
+        """
+        AppBuilder constructor
```
Comment (uranusjr) [inline airflow/www/extensions/init_appbuilder.py]:
```suggestion
        App-builder constructor.

```

File: airflow/www/extensions/init_appbuilder.py [hunk 4]
```diff
@@ -15,11 +15,627 @@
...
+            template_folder="templates",
+            static_folder=self.static_folder,
+            static_url_path=self.static_url_path,
+        )
+        self.get_app.register_blueprint(bp)
+
+    def _add_admin_views(self):
+        """
+        Registers indexview, utilview (back function), babel views and Security views.
+        """
```
Comment (uranusjr) [inline airflow/www/extensions/init_appbuilder.py]:
```suggestion
        """Register indexview, utilview (back function), babel views and Security views."""
```

File: airflow/www/extensions/init_appbuilder.py [hunk 5]
```diff
@@ -15,11 +15,627 @@
...
+        self.add_view_no_menu(UtilView())
+        self.bm.register_views()
+        self.sm.register_views()
+        self.openapi_manager.register_views()
+        self.menuapi_manager.register_views()
+
+    def _add_addon_views(self):
+        """
+        Registers declared addon's
+        """
```
Comment (uranusjr) [inline airflow/www/extensions/init_appbuilder.py]:
```suggestion
        """Register declared addons."""
```

File: airflow/www/extensions/init_appbuilder.py [hunk 6]
```diff
@@ -15,11 +15,627 @@
...
+        href="",
+        icon="",
+        label="",
+        category="",
+        category_icon="",
+        category_label="",
+        menu_cond=None,
+    ):
+        """
+        Add your views associated with menus using this method.
```
Comment (uranusjr) [inline airflow/www/extensions/init_appbuilder.py]:
```suggestion
        """Add your views associated with menus using this method.

```

File: airflow/www/extensions/init_appbuilder.py [hunk 7]
```diff
@@ -15,11 +15,627 @@
...
+            optional, your override for the global static folder
+        :param static_url_path:
+            optional, your override for the global static url path
+        :param security_manager_class:
+            optional, pass your own security manager class
+        :param update_perms:
+            optional, update permissions flag (Boolean) you can use
+            FAB_UPDATE_PERMS config key also
+        """
+
```
Comment (uranusjr) [inline airflow/www/extensions/init_appbuilder.py]:
```suggestion
```

**Review by uranusjr — COMMENTED** [review]
Our linter has some complaints on FAB’s coding style :p

File: airflow/www/extensions/init_appbuilder.py [hunk 8]
```diff
@@ -15,11 +15,623 @@
...
+        return reduce(getattr, tmp[1:], package)
+    except Exception as e:
+        log.exception(e)
+        log.error(LOGMSG_ERR_FAB_ADDON_IMPORT.format(class_path, e))
+
+
+class AirflowAppBuilder:
+    """
+    This is the base class for all the framework.
+    This is were you will register all your views
```
Comment (josh-fell) [inline airflow/www/extensions/init_appbuilder.py]:
```suggestion
    This is where you will register all your views
```

File: airflow/www/extensions/init_appbuilder.py [hunk 9]
```diff
@@ -15,11 +15,623 @@
...
+    and create the menu structure.
+    Will hold your flask app object, all your views, and security classes.
+    initialize your application like this for SQLAlchemy::
```
Comment (josh-fell) [inline airflow/www/extensions/init_appbuilder.py]:
```suggestion
    Initialize your application like this for SQLAlchemy::
```
No strong opinion here really. Just makes it consistent with the "When using MongoEngine::..." portion of this docstring.

Comment (ashb) [reply airflow/www/extensions/init_appbuilder.py]: These are all as-is from Flask-AppBuilder and in this PR we are pulling it all in without changes.

So yes, but not in this PR.

Comment (josh-fell) [reply airflow/www/extensions/init_appbuilder.py]: Right on, thanks Ash. @jhtimmins Ignore me.
